src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -224,8 +227,95 @@ package experimental {
         case _ => false
       }
     }
+
+    /**
+     * Calculate the inverse of this pattern set.
+     * 
+     * @return A BitSet matching all value (of the given with) iff it doesn't match this pattern.
+     */
+    def inverse: BitSet = {
+      val total = BitPat("b" + ("?".repeat(this.getWidth)))
+      total.subtract(this)
+    }
```

Comment:
It may run forever, but this API is necessary, I think the time complexity may need to be documented? or at least tell users this API is time consuming.

Reply:
This is just a simple reduction into `BitSet.subtract`. If I'm not mistaken, the maximum number of generated terms is width * len(term), which doesn't seems that bad :tm:

Actually, if we are to document about the complexity, we may also want to document about the complexity of `BitSet.subtract`?

Reply:
yes, basically all complexity should come from `subtract`. emmm. that's time consuming as well. I think mention it on ScalaDoc is good enough yet.